libflash/platformfs.cpp

```diff
@@ -231,7 +231,9 @@ ExpectedBool mender::io::IsUBIDevice(const string &path) {
 
 Error mender::io::SetUbiUpdateVolume(File f, size_t size) {
 	errno = 0;
-	ioctl(f, UBI_IOCVOLUP, &size);
+	// Argument is 64-bit, even on 32-bit platforms.
+	int64_t size64 = size;
```

Comment:
Nit: `int64_t size64 {size};`

Reply:
Or cast in the function

Reply:
I don't think the function should take that argument. We should restrict it to be as platform specific as possible (one could maybe argue that `size_t` should be 64-bit already, but that's a different discussion).

Reply:
> Nit: `int64_t size64 {size};`

This made it much uglier, since brace initialization is more restrictive and we now have to cast it. But I suppose that's a good thing here, because it highlights what we are doing. So all in all: Good suggestion!